machine_learning_hep/utilities.py

```diff
@@ -448,26 +448,67 @@ def get_plot_range(val_min, val_max, margin_min, margin_max, logscale=False):
         val_max_plot = val_max + k_max * val_range
     return val_min_plot, val_max_plot
 
-def get_y_window_gr(l_gr: list):
-    '''Return the minimum and maximum value so that all the points of the graphs in the list
-    fit in the range including the error bars.'''
+def get_x_window_gr(l_gr: list, with_errors=True):
+    '''Return the minimum and maximum x value so that all the points of the graphs in the list
+    fit in the range (by default including the error bars).'''
+    def err_low(graph):
+        return graph.GetEXlow if isinstance(graph, TGraphAsymmErrors) else graph.GetEX
+    def err_high(graph):
+        return graph.GetEXhigh if isinstance(graph, TGraphAsymmErrors) else graph.GetEX
+
     if not isinstance(l_gr, list):
         l_gr = [l_gr]
-    y_min = min([min([(gr.GetY())[i] - (gr.GetEYlow())[i] \
-        for i in range(gr.GetN())]) for gr in l_gr])
-    y_max = max([max([(gr.GetY())[i] + (gr.GetEYhigh())[i] \
-        for i in range(gr.GetN())]) for gr in l_gr])
+    x_min = float("inf")
+    x_max = float("-inf")
+    for gr in l_gr:
+        for i in range(gr.GetN()):
+            x_min = min(x_min, (gr.GetX())[i] - ((err_low(gr)())[i] if with_errors else 0))
+            x_max = max(x_max, (gr.GetX())[i] + ((err_high(gr)())[i] if with_errors else 0))
+    return x_min, x_max
+
+def get_x_window_his(l_his: list):
+    '''Return the minimum and maximum x value so that all the bins of the histograms in the list
+    fit in the range.'''
+    if not isinstance(l_his, list):
+        l_his = [l_his]
+    x_min = float("inf")
+    x_max = float("-inf")
+    for his in l_his:
+        x_min = min(x_min, his.GetXaxis().GetBinLowEdge(1))
+        x_max = max(x_max, his.GetXaxis().GetBinUpEdge(his.GetNbinsX()))
+    return x_min, x_max
+
+def get_y_window_gr(l_gr: list, with_errors=True):
+    '''Return the minimum and maximum y value so that all the points of the graphs in the list
+    fit in the range (by default including the error bars).'''
+    def err_low(graph):
+        return graph.GetEYlow if isinstance(graph, TGraphAsymmErrors) else graph.GetEY
+    def err_high(graph):
+        return graph.GetEYhigh if isinstance(graph, TGraphAsymmErrors) else graph.GetEY
+
+    if not isinstance(l_gr, list):
+        l_gr = [l_gr]
+    y_min = float("inf")
+    y_max = float("-inf")
+    for gr in l_gr:
+        for i in range(gr.GetN()):
+            y_min = min(y_min, (gr.GetY())[i] - ((err_low(gr)())[i] if with_errors else 0))
+            y_max = max(y_max, (gr.GetY())[i] + ((err_high(gr)())[i] if with_errors else 0))
     return y_min, y_max
 
-def get_y_window_his(l_his: list):
-    '''Return the minimum and maximum value so that all the points of the histograms in the list
-    fit in the range including the error bars.'''
+def get_y_window_his(l_his: list, with_errors=True):
```

Comment:
Yep.

Reply:
Because of the previous absence of the `make_plot` function, `get_y_window_his` and `get_y_window_gr` are used many times in `analyzer_jet.py` so I would not change them for now (until a more relaxed period).